### PR TITLE
Security Improvements in Load balancer

### PR DESCRIPTION
**Pull request by kalroy**
Automate Load Balancer needs to include provide several security improvements.

Signed-off-by: Kallol Roy <karoy@progress.com>

### :nut_and_bolt: Description: What code changed, and why?

Automate Load Balancer misses a bunch of headers to be set for the secure data transfer and security of the Automate UI.

### :chains: Related Resources
#4765
#4851
#4852 
#4857
#4858 
#4859 
#4860  

### :+1: Definition of Done

Added Strict-Transport-Security, x-xss-protection, x-content-type-options  in Response header

### :athletic_shoe: How to Build and Test the Change
-  `rebuild components/automate-load-balancer`
- `start_all_services`
- 4765: hit /automate.conf.js. Check the header to see if the Strict-Transport-Security header is set.
- 4851: hit http://a2-dev.test and check the 301 redirection for X-XSS-Protection header
- 4852: Hit http://a2-dev.test and check the response for X-Content-Type-Options header
- 4857: hit /automate.conf.js. Check the header to see if X-Content-Type-Options header is set
- 4858: hit /automate.conf.js. Check the header to see if X-XSS-Protection header is set
- 4859: hit /automate.conf.js. Check the header to see if the Strict-Transport-Security header is set.
- 4860: same as the upper one

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 1]
```diff
@@ -154,6 +157,8 @@ http {
     add_header x-xss-protection "1; mode=block" always;
     add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
     add_header X-Content-Type-Options "nosniff" always;
+    # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:160]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 2]
```diff
@@ -177,6 +182,11 @@ http {
 
     location /automate.conf.js {
       add_header Cache-Control "private, no-cache, no-store";
+      add_header x-xss-protection "1; mode=block" always;
+      add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header X-Content-Type-Options "nosniff" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:189]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 3]
```diff
@@ -193,13 +203,23 @@ http {
       proxy_pass https://automate-gateway;
       # Required to make persistent connections happen
       proxy_set_header Connection "";
+      add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header X-Content-Type-Options "nosniff" always;
+      add_header x-xss-protection "1; mode=block" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:210]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 4]
```diff
@@ -193,13 +203,23 @@ http {
...
     location /apis/ {
       proxy_set_header X-Client-Cert $ssl_client_escaped_cert;
       proxy_pass https://automate-gateway;
       # Required to make persistent connections happen
       proxy_set_header Connection "";
+      add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header X-Content-Type-Options "nosniff" always;
+      add_header x-xss-protection "1; mode=block" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:222]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 5]
```diff
@@ -208,13 +228,23 @@ http {
       proxy_pass https://automate-gateway/api/v0/events/data-collector;
       # Required to make persistent connections happen
       proxy_set_header Connection "";
+      add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header X-Content-Type-Options "nosniff" always;
+      add_header x-xss-protection "1; mode=block" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:235]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 6]
```diff
@@ -208,13 +228,23 @@ http {
...
     location /compliance/profiles/ {
       proxy_set_header X-Client-Cert $ssl_client_escaped_cert;
       proxy_pass https://automate-gateway/api/v0/compliance/profiles/; # TODO does this append? Is this tested?
       # Required to make persistent connections happen
       proxy_set_header Connection "";
+      add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header X-Content-Type-Options "nosniff" always;
+      add_header x-xss-protection "1; mode=block" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:247]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 7]
```diff
@@ -227,7 +257,9 @@ http {
       add_header Cache-Control "no-cache, no-store, must-revalidate";
       add_header x-xss-protection "1; mode=block" always;
       add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:260]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 8]
```diff
@@ -237,6 +269,8 @@ http {
       add_header Cache-Control "no-cache, no-store, must-revalidate";
       add_header x-xss-protection "1; mode=block" always;
       add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:273]: Remove this commented line

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 9]
```diff
@@ -250,6 +284,8 @@ http {
       add_header Cache-Control "private"; # allow caching, but no sharing of cached data
       add_header x-xss-protection "1; mode=block" always;
       add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      # add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:288]: Remove this commented lines

File: components/automate-load-balancer/habitat/config/nginx.conf [hunk 10]
```diff
@@ -260,6 +296,8 @@ http {
       add_header Cache-Control "private"; # allow caching, but no sharing of cached data
       add_header x-xss-protection "1; mode=block" always;
       add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+      #add_header Content-Security-Policy "default-src 'self';";
+      # add_header Content-Security-Policy "frame-ancestors 'self';";
```
Comment (kalroy) [inline components/automate-load-balancer/habitat/config/nginx.conf:300]: Remove this commented lines

**Review by kalroy — COMMENTED** [review]
Need to remove these changes

File: components/automate-load-balancer/habitat/config/nginx.conf (hunk 2 above)
Comment (Venkatesh-rengasamy) [reply components/automate-load-balancer/habitat/config/nginx.conf]: Its done.

File: components/automate-load-balancer/habitat/config/nginx.conf (hunk 4 above)
Comment (Venkatesh-rengasamy) [reply components/automate-load-balancer/habitat/config/nginx.conf]: Its done.

File: components/automate-load-balancer/habitat/config/nginx.conf (hunk 6 above)
Comment (Venkatesh-rengasamy) [reply components/automate-load-balancer/habitat/config/nginx.conf]: Its done.

File: components/automate-load-balancer/habitat/config/nginx.conf (hunk 8 above)
Comment (Venkatesh-rengasamy) [reply components/automate-load-balancer/habitat/config/nginx.conf]: Its done.

File: components/automate-load-balancer/habitat/config/nginx.conf (hunk 10 above)
Comment (Venkatesh-rengasamy) [reply components/automate-load-balancer/habitat/config/nginx.conf]: Its done.

**Review by Venkatesh-rengasamy — APPROVED** [review]
Looks good to me.

**Review by punitmundra — APPROVED** [review]

**Pull request merged**